doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+- IndexIdentifierMap
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size. The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers). The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces, a mapping is required from policy index to local and global identifiers. This mapping is provided by this map.
+- IdentifierDescriptionMap
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS. This map maps a local policy identifier to its corresponding description.
+- MPLSMap
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to enforce the policy in the internal network. The MPLSMap data structure stores a mapping between policy indices and the integer MPLS label that is applied to the packet at the border router. Routers periodically fetch this map from the control service. A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- RemotePolicyCache
+    When a local policy is queried at a remote AS, the resulting policy description is cached at the requesting AS’ FABRID Manager, such that subsequent requests can be served from cache.
+
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The IndexIdentifierMap and SupportedIndicesMap are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
+Exposing policies to the end hosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+This results in a set of PolicyIdentifiers per hop, which can then be used by the application, such as with the usage of a
+specific ‘sequence’ parameter which incorporates the policies.
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
```

Comment:
Should we maybe outline the YAML policy, what options there are etc?

Reply:
I think it would be good if you could provide an example on how it could be configured

Reply:
@jeltevanbommel Could you please provide such an example?

Reply:
See below